cpp/recognition/CographRecognition.cpp

```diff
@@ -0,0 +1,58 @@
+#include <list>
+#include "graph/GraphTools.hpp"
+#include "recognition/CorneilStewartPerlCographRecognition.h"
+
+namespace Koala {
+    CorneilStewartPerlCographRecognition::CorneilStewartPerlCographRecognition(NetworKit::Graph &graph) : graph(graph),
+                                                                                                          is_cograph(
+                                                                                                                  State::UNKNOWN) {
+
+    }
+
+    bool CorneilStewartPerlCographRecognition::isCograph() const {
+        assureFinished();
+        return is_cograph == State::COGRAPH;
+    }
+
+    CorneilStewartPerlCographRecognition::State CorneilStewartPerlCographRecognition::getState() const {
+        assureFinished();
+        return is_cograph;
+    }
+
+    CorneilStewartPerlCographRecognition::State recognition(NetworKit::Graph &graph) {
+        return CorneilStewartPerlCographRecognition::Cograph_Recognition(graph);
+    }
+
+    void CorneilStewartPerlCographRecognition::run() {
+        hasRun = true;
+        if (is_cograph != State::UNKNOWN) {
+            return;
+        }
+        is_cograph = recognition(graph);
+        if (is_cograph != State::UNKNOWN) {
+            return;
+        }
+        is_cograph = State::COGRAPH;
+    }
+
+    void CorneilStewartPerlCographRecognition::check() const {
```

Comment:
As I suggested above, please move it to CographRecognition class - as it's independent of particular recognition algorithms.

Reply:
fixed